### PR TITLE
Support hot-deploying (with graceful shutdown)

### PR DESCRIPTION
**Pull request by tcnksm**
For hot-deploying, support `lestrrat/go-server-starter` listeners. 

Graceful shutdown is done by Go1.8's graceful shutdown feature (see [release note](https://tip.golang.org/doc/go1.8#http_shutdown)). So this PR must be merged after go1.8 release. 

## Test

To test hot-deploying feature, run `wbt` by `server-starter` and kill HUP to it constantly (per sec.) and send 1000 requests to check there is no lost requests. 

To kick `wbt`,

```bash
$ start_server --port 8080 --pid-file=/tmp/test.pid -- wbt -c config/example.toml
```

To constantly kill HUP it,

```bash
$ while true; do kill -HUP $(cat /tmp/test.pid); sleep 1s; done
```

Requests are done by [`hey`]() command,

```bash
$ hey -n 1000 -c 100 -D test.json -m POST 'http://localhost:8080/wbt'
```

The results are the followings. Sometimes requests takes long time (but very small number) but no requests are lost.

```bash
Summary:
  Total:        25.3181 secs
  Slowest:      19.4102 secs
  Fastest:      0.0004 secs
  Average:      1.6511 secs
  Requests/sec: 39.4974
  Total data:   564877 bytes
  Size/request: 564 bytes

Status code distribution:
  [200] 1000 responses

Response time histogram:
  0.000 [1]     |
  1.941 [623]   |∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  3.882 [259]   |∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  5.823 [86]    |∎∎∎∎∎∎
  7.764 [6]     |
  9.705 [0]     |
  11.646 [0]    |
  13.587 [0]    |
  15.528 [0]    |
  17.469 [0]    |
  19.410 [25]   |∎∎

Latency distribution:
  10% in 0.0029 secs
  25% in 0.0049 secs
  50% in 0.0309 secs
  75% in 1.9808 secs
  90% in 5.0027 secs
  95% in 5.0092 secs
  99% in 19.4014 secs
```

### CODE REVIEW

File: config/example.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Port = "29300"
-LogLevel = "error"
+LogLevel = "debug"
```
Comment (cubicdaiya) [inline config/example.toml]: Leave LogLevel **error**.

File: widebullet.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 const (
-	Version = "0.3.0"
+	Version = "0.3.1"
```
Comment (cubicdaiya) [inline widebullet.go]: Leave Version **0.3.0**. We'll update version when latest version is released.

File: cmd/wbt/wbt.go [hunk 3]
```diff
@@ -36,6 +47,35 @@ func main() {
...
+		}
+	}()
+
+	// Watch SIGTERM signal and then gracefully shutdown
+	sigCh := make(chan os.Signal)
+	signal.Notify(sigCh, syscall.SIGTERM)
+	<-sigCh
+
+	wbt.EL.Out(wlog.Debug, "Start to shutdown server")
+	ctx, cancel := context.WithTimeout(context.Background(), defaultShutdownTimeout)
```
Comment (cubicdaiya) [inline cmd/wbt/wbt.go]: How about making **ShutdownTimeout** configurable?

**Comment by cubicdaiya** [comment]
Please update [Requirements](https://github.com/mercari/widebullet#requirement).

File: config/example.toml (hunk 1 above)
Comment (tcnksm) [reply config/example.toml]: Backed

File: widebullet.go (hunk 2 above)
Comment (tcnksm) [reply widebullet.go]: Backed

File: cmd/wbt/wbt.go (hunk 3 above)
Comment (tcnksm) [reply cmd/wbt/wbt.go]: Chage it configurable !

**Comment by tcnksm** [comment]
And update documentation !

**Pull request closed without merging**